rfcs/011-graphql.md

```diff
@@ -0,0 +1,81 @@
+# RFC 011 - Best Practices with GraphQL
+
+## Affected projects
+
+- Frontend
+
+## Motivation
+
+It's not always clear what are the best practices when using GraphQL in our frontend codebase. This document aims to help with that.
+
+## Best Practices
+
+### Operation Name
+
+When troubleshooting and observing GraphQL usage, it's nice to have great names so we can clearly identify what is the operation and wehere it's coming from.
+
+
+All GraphQL queries and mutations should have a clear "Operation Name" that is unique within the project.
+
+- It should be in PascalCase
+- Should not contain `Query` or `Mutation` suffix.
+- Should not contain `get` or `fetch` prefix.
+
+Good
+
+- `query LoggedInUser`
+- `mutation CreateCollective`
+
+### Variable name
+
+All GraphQL queries and mutations should been assigned to a variable (usually a JavaScript `const`).
+
+In most of the cases, it should be named like the "Operation Name", but:
+
+- Like all variables in the codebase, it should be in camelCase.
```

Comment:
In [their documentation](https://www.apollographql.com/docs/react/data/queries/), Apollo always use uppercase because these are constants variables that are not meant to be mutated (ie. `CREATE_COLLECTIVE_MUTATION`). I think it's a better pattern, but I'm fine with camelCase.

Reply:
Yeah, I hesitated and considered that. I like the idea of following Apollo patterns.

However, I felt we were moving too far from our own coding guidelines.